### PR TITLE
Redo auth

### PR DESCRIPTION
**Pull request by dannymandel**
iSB side of https://github.com/isamplesorg/isamples_webui/issues/81

### CODE REVIEW

File: isb_web/config.py [hunk 1]
```diff
@@ -18,36 +18,6 @@ class Settings(BaseSettings):
     # e.g. http://localhost:8983/solr/isb_core_records/
     solr_url: str = "UNSET"
 
-    # OAuth client id, generated by OAuth provider
```
Comment (dannymandel) [inline isb_web/config.py:21]: Remove unused github oauth

File: isb_web/config.py [hunk 2]
```diff
@@ -79,6 +49,10 @@ class Settings(BaseSettings):
     # This must not ever be checked in.  Set it by doing export ORCID_CLIENT_SECRET=foobar
     orcid_client_secret: str = ""
 
+    orcid_issuer: str = "https://sandbox.orcid.org"
+
+    secret_key: str = "123456"
```
Comment (dannymandel) [inline isb_web/config.py]: @datadavev per discussion on this morning's call we will leave this checked in.  I will give it a better name

File: isb_web/main.py [hunk 3]
```diff
@@ -54,42 +54,6 @@
     pass
 L = logging.getLogger("ISB")
 
-# Cookie chaff
```
Comment (dannymandel) [inline isb_web/main.py:57]: More removal of unused GitHub oauth

File: isb_web/main.py [hunk 4]
```diff
@@ -102,16 +66,9 @@
...
 dao = SQLModelDAO(None)
 
-app.add_middleware(
-    fastapi.middleware.cors.CORSMiddleware,
-    allow_origins=["*"],
-    allow_methods=["*"],
-    allow_headers=["*"],
-)
-app.add_middleware(SessionMiddleware, secret_key=COOKIE_SECRET)
-
```
Comment (dannymandel) [inline isb_web/main.py:114]: For future reference, the middleware of the main app *does* affect any registered sub apps!

File: mypy.ini [hunk 5]
```diff
@@ -32,4 +32,12 @@ ignore_missing_imports = True
...
 ignore_missing_imports = True
 [mypy-jsonschema.*]
+ignore_missing_imports = True
+[mypy-starlette_oauth2_api.*]
+ignore_missing_imports = True
+[mypy-authlib.*]
+ignore_missing_imports = True
+[mypy-authlib.integrations.*]
+ignore_missing_imports = True
+[mypy-authlib.integrations.starlette_client.*]
```
Comment (dannymandel) [inline mypy.ini:42]: Unfortunately it looks like none of these have type hints, so we need to tell mypy to ignore them.

**Comment by dannymandel** [comment]
Will resolve any feedback in followup PRs.  Going to merge this so we can deploy to mars and test it all out.

File: isb_web/manage.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+@manage_api.get("/logout")
+async def logout(request: starlette.requests.Request):
+    """
+    Logout by removing the cookie from the user session.
+
+    Note that this does not invalidate the JWT, which could continue
+    to be used. That's a "feature" of JWTs.
+    """
+    request.session.pop("user", None)
+    return starlette.responses.RedirectResponse(url="/")
```
Comment (qgan7125) [inline isb_web/manage.py:221]: Should this endpoint redirect to `/ui/`?

**Pull request merged**